packages/patrol/android/src/main/kotlin/pl/leancode/patrol/Automator.kt

```diff
@@ -334,29 +315,28 @@ class Automator private constructor() {
 
         val notifications = mutableListOf<Contracts.Notification>()
         for (notificationContainer in notificationContainers) {
-            val notification = notification {
-                val appName = notificationContainer.findObject(By.res("android:id/app_name_text"))?.text
-                if (appName != null) {
-                    this.appName = appName
-                }
+            val appName = notificationContainer.findObject(By.res("android:id/app_name_text"))?.text
 
-                val title = notificationContainer.findObject(By.res("android:id/title"))?.text
-                    ?: notificationContainer.findObject(By.res("com.android.systemui:id/notification_title"))?.text
-                if (title != null) {
-                    this.title = title
-                } else {
-                    Logger.e("Could not find title text")
-                }
+            val content = notificationContainer.findObject(By.res("android:id/text"))?.text
+                ?: notificationContainer.findObject(By.res("android:id/big_text"))?.text
+                ?: notificationContainer.findObject(By.res("com.android.systemui:id/notification_text"))?.text
+            if (content == null) {
+                Logger.e("Could not find content text")
+            }
 
-                val content = notificationContainer.findObject(By.res("android:id/text"))?.text
-                    ?: notificationContainer.findObject(By.res("android:id/big_text"))?.text
-                    ?: notificationContainer.findObject(By.res("com.android.systemui:id/notification_text"))?.text
-                if (content != null) {
-                    this.content = content
-                } else {
-                    Logger.e("Could not find content text")
-                }
+            val title = notificationContainer.findObject(By.res("android:id/title"))?.text
+                ?: notificationContainer.findObject(By.res("com.android.systemui:id/notification_title"))?.text
+            if (title == null) {
+                Logger.e("Could not find title text")
             }
+
+            val notification = Contracts.Notification(
+                appName = appName,
+                content = content ?: "",
+                title = title ?: "",
+                raw = "" //  Should it be optional ?
```

Comment:
I think it's only set on iOS to workaround some limitation there. Hacky :(

Setting it to "Not available on Android" is good enough for now, I guess.

Reply:
In a private discussion, we decided to make this field optional and keep it consistent with the other fields.